packages/teleterm/src/mainProcess/runtimeSettings.ts

```diff
@@ -209,4 +211,32 @@ function getUnixSocketNetworkAddress(socketName: string) {
   return `unix://${path.resolve(app.getPath('userData'), socketName)}`;
 }
 
+const UUID_V4_REGEX =
+  /^[0-9A-F]{8}-[0-9A-F]{4}-4[0-9A-F]{3}-[89AB][0-9A-F]{3}-[0-9A-F]{12}$/i;
+
+function loadInstallationId(): string {
+  const filePath = path.resolve(app.getPath('userData'), 'installation_id');
+  const writeNewId = (): string => {
```

Comment:
This is more of a personal preference I guess but this function could have been a regular top-level function which accepts path as an argument. 😶‍🌫️

For some reason, I find the pattern of defining ad-hoc functions like this somewhat counterintuitive. It's kinda similar to this common approach where you cannot really follow the execution of the function top to bottom:

```typescript
const onSuccess = (foo, bar) => { … }
const onError = (error) => { … }

something(baz, onSuccess, onError)
```

I know that React kinda popularized this approach, but most of the time those ad-hoc functions you define in React are simple event handlers where you don't really care about the side effects or they just render HTML. Here they directly impact the execution path of the whole function.

Reply:
I thought that `writeNewId` makes sense only in a context of `loadInstallationId` so I wanted to make it available only for that function. I could create a new file for it, but these functions did not seem complex enough to justify it. 
But I'm fine with converting `writeNewId` to a top-level function.

Reply:
> I could create a new file for it, but these functions did not seem complex enough to justify it.

I'd say in situations like that it's more than fine to do it! `runtimeSettings` is already complex enough so abstracting all this stuff in a separate file is good IMHO.

Reply:
Done!